pkg/commands/git_commands/branch.go

```diff
@@ -197,7 +197,7 @@ func (self *BranchCommands) IsHeadDetached() bool {
 
 func (self *BranchCommands) Rename(oldName string, newName string) error {
 	cmdArgs := NewGitCmd("branch").
-		Arg("--move", oldName, newName).
+		Arg("--move", oldName, strings.ReplaceAll(newName, " ", "-")).
```

Comment:
We have [a function](https://github.com/jesseduffield/lazygit/blob/a3cd1e93be25517657d091e7ba7be0236768425f/pkg/gui/controllers/helpers/refs_helper.go#L188) for this in `refs_helper.go`. It's private, so you can't access it from here, but that's easy to change by uppercasing it.

Even though the function is trivial, it's still worth having it (and using it from both places), because we might add more sanitizations in the future, and then we want them to be the same for both.

Reply:
I believe I have done it 😅. I've never used golang before and writing an integration test looks a little above my abilities. Would be great to have yours pushed in and for me to have a look at it 🙂